homeassistant/components/imap_email_content/sensor.py

```diff
@@ -214,15 +214,18 @@ def get_msg_text(email_message):
         message_untyped_text = None
 
         for part in email_message.walk():
-            if part.get_content_type() == CONTENT_TYPE_TEXT_PLAIN:
-                if message_text is None:
-                    message_text = part.get_payload()
-            elif part.get_content_type() == "text/html":
-                if message_html is None:
-                    message_html = part.get_payload()
-            elif part.get_content_type().startswith("text"):
-                if message_untyped_text is None:
-                    message_untyped_text = part.get_payload()
+            if (
+                part.get_content_type() == CONTENT_TYPE_TEXT_PLAIN
+                and message_text is None
+            ):
+                message_text = part.get_payload()
+            elif part.get_content_type() == "text/html" and message_html is None:
```

Comment:
We'll now check this case if the above first check is true but the above second check is false. We didn't do that with the old pattern. I don't know if this matters but the logic has changed.

Reply:
Adjusted.